### ISSUE TITLE
[Bug]: Memory leak in the storage deserilize 

### ISSUE THREAD

**Issue by godchen0212** [milvus-io/milvus#12585]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Environment

```markdown
- Milvus version: master(26405d1)
- Deployment mode(standalone or cluster):both
- SDK version(e.g. pymilvus v2.0.0rc2):2.0.0rc9.dev22
- OS(Ubuntu or CentOS): Ubuntu 20.04
- CPU/Memory: i7-8700  64GB
- GPU: GTX 1070
- Others:
```

### Current Behavior

I test it on my local machine and go version is 1.16. I test insert 500K data with dim 128 in to milvus, load and release for 100 times. The memory usage increase from 58M to 1.5G and fell back to 700M after two minutes.

### Expected Behavior

The memory usage should fell back to 100M

### Steps To Reproduce

```markdown
import random

from pymilvus import (
    connections, list_collections,
    FieldSchema, CollectionSchema, DataType,
    Collection,
    utility
)


def hello_milvus():
    # create connection
    connections.connect(host="127.0.0.1",port="19530")

    print(f"\nList collections...")
    print(list_collections())

    # create collection
    dim = 128
    default_fields = [
        FieldSchema(name="count", dtype=DataType.INT64, is_primary=True),
        FieldSchema(name="random_value", dtype=DataType.DOUBLE),
        FieldSchema(name="float_vector", dtype=DataType.FLOAT_VECTOR, dim=dim)
    ]
    default_schema = CollectionSchema(fields=default_fields, description="test collection")

    print(f"\nCreate collection...")
    collection = Collection(name="hello_milvus", schema=default_schema)

    print(f"\nList collections...")
    print(list_collections())

    #  insert data
    nb = 500000
    vectors = [[random.random() for _ in range(dim)] for _ in range(nb)]
    collection.insert(
        [
            [i for i in range(nb)],
            [float(random.randrange(-20, -10)) for _ in range(nb)],
            vectors
        ]
    )

    print(f"\nGet collection entities...")
    print(collection.num_entities)

    import time
    try_time = 100
    while(try_time>0):
        print(f"\nload collection...", try_time)
        collection.load()

        print(f"\nrelease collection...")
        collection.release()
        try_time-=1

hello_milvus()
```

**Comment by godchen0212** [comment]
I pull the source code and test arrow library, Modified their example code for testing, and found that the arrow library originally had this problem. So I filed an issue for them, the link is [Arrow-14987](https://issues.apache.org/jira/browse/ARROW-14987).

A brief description is as follows, there are chunk settings in columnar storage. For each chunk, memory needs to be allocated in memory to record meta. The default chunk size is 3. Since our system is more biased towards columnar storage and columnar reading, a chunk size of 3 will result in the storage of meta information even larger than Data storage, this is the fundamental problem of memory growth.

There are two solutions, one is to use jemalloc, and the other is to increase the chunk size. In fact, we have already used a larger chunk size in our code. I suggest using jemalloc to try to optimize this problem.

**Comment by godchen0212** [comment]
Issue #12893 optimizes some problems by release unused memory in parquet memory pool.

**Comment by godchen0212** [comment]
PR #13029 decreases memory usage from 470M to 426M by make arrowusing jemalloc.

**Comment by godchen0212** [comment]
I subsequently used the simplified repo of @czs007  to do the following experiments:
1. In order to eliminate the problem of PayloadStream.cpp that we wrote, I applied PayloadStream.cpp to the parquet example, changed the original read file to []byte, and passed it to construct PayloadInputStream object. As a result, there is no memory leak, indicating that there is no problem with PayloadStream.cpp.
2. In order to eliminate the issue of whether the incoming buffer is released, I added a new function NewPayloadReaderSecond() to the repo. It has no parameters. It reads the file directly from the file and saves it as a local variable []byte, and builds PayloadInputStream. Then The creation of the buffer is annotated externally, and the  memory leak not happens.
3. In order to reproduce the problem in 2, I created a buffer, but did not pass in the function, and found that the problem reproduced. I suspect it is the problem of partially reading the buffer in go.
4. I directly canceled call of c part, and only used the function of the go part to read the file, and found memory leaks to 100M. The specific go code is as follows:
`
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"runtime"
	"time"
)

func execute(i int, enableGC bool) {
	fmt.Println("round ", i)
	b, err := ioutil.ReadFile("parquet-arrow-example.parquet") // just pass the file name
	if err != nil {
		fmt.Print(err)
	}
	fmt.Println(len(b))
}

func main() {
	if len(os.Args) < 1 {
		_, _ = fmt.Fprint(os.Stderr, "usage: my_example --enableGC\n")
		return
	}

	for i := 0; i < 100; i++ {
		execute(i, false)
	}
	runtime.GC()
	time.Sleep(1000 * time.Second)
}

`
